language-support/java/codegen/src/test/scala/com/digitalasset/daml/lf/codegen/backend/java/FromValueSpec.scala

```diff
@@ -16,15 +16,17 @@ final class FromValueSpec extends AnyWordSpec with Matchers {
       val fromConstructor: ParameterizedContractId[Bar] =
         new ParameterizedContractId(new Bar.ContractId("SomeID"))
       val parametrizedContractId: ParameterizedContractId[Bar] =
-        ParameterizedContractId.fromValue(fromConstructor.toValue(_.toValue), Bar.fromValue)
+        ParameterizedContractId
+          .fromValue(Bar.fromValue())
+          .fromValue(fromConstructor.toValue(_.toValue))
       val contractIdBar: ContractId[Bar] = parametrizedContractId.parameterizedContractId
 
       illTyped(
         "contractIdBar: Bar.ContractId",
         "type mismatch.+ContractId\\[.+Bar\\].+Bar.ContractId",
       )
 
-      contractIdBar should not be a[Bar.ContractId]
+      contractIdBar shouldBe a[Bar.ContractId]
```

Comment:
This is the whole point for contractId to be in the value decoder interface here